Content.Server/GameObjects/Components/Weapon/Melee/FlashComponent.cs

```diff
@@ -130,20 +134,12 @@ private void Flash(IEntity entity, IEntity user)
         // TODO: Check if target can be flashed (e.g. things like sunglasses would block a flash)
         private void Flash(IEntity entity, IEntity user, int flashDuration)
         {
-            if (entity.TryGetComponent(out ServerOverlayEffectsComponent overlayEffectsComponent))
+            if (entity.TryGetComponent(out IActorComponent actor))
             {
-                if (!overlayEffectsComponent.TryModifyOverlay(nameof(SharedOverlayID.FlashOverlay),
-                    overlay =>
-                    {
-                        if (overlay.TryGetOverlayParameter<TimedOverlayParameter>(out var timed))
-                        {
-                            timed.Length += flashDuration;
-                        }
-                    }))
-                {
-                    var container = new OverlayContainer(SharedOverlayID.FlashOverlay, new TimedOverlayParameter(flashDuration));
-                    overlayEffectsComponent.AddOverlay(container);
-                }
+                var netMgr = IoCManager.Resolve<IServerNetManager>();
+                var netMessage = netMgr.CreateNetMessage<MsgFlash>();
+                netMessage.TimeMilliseconds = flashDuration;
+                netMgr.ServerSendMessage(netMessage, actor.playerSession.ConnectedClient);
```

Comment:
flash should not have its own message, it should use a (preferably) state or a component message.

Reply:
This became necessary because for whatever reason Flashable didn't apply to this specific flash. The comment on FlashWhiteOverlay suggests they weren't supposed to be different in the first place, which would solve both problems.